example/oven_fsm.dart

```diff
@@ -0,0 +1,228 @@
+/// Copyright (C) 2023 Intel Corporation
+/// SPDX-License-Identifier: BSD-3-Clause
+///
+/// oven_fsm.dart
+/// A simple oven FSM implementation using ROHD.
+///
+/// 2023 February 13
+/// Author: Yao Jing Quek <yao.jing.quek@intel.com>
+///
+
+// ignore_for_file: avoid_print
+
+// Import the ROHD package.
+import 'package:rohd/rohd.dart';
+
+// Import the counter module implement in example.dart.
+import './example.dart';
+
+// Enumerated type named `OvenState` with four possible states:
+// `standby`, `cooking`,`paused`, and `completed`.
+enum OvenState { standby, cooking, paused, completed }
+
+// One-hot encoded `Button` that extends from Const() value.
+// Represent start, pause, and resume as constant value `00`, `01`,
+// and `10` respectively.
+class Button extends Const {
+  Button._(int super.value) : super(width: 2);
+  Button.start() : this._(bin('00'));
+  Button.pause() : this._(bin('01'));
+  Button.resume() : this._(bin('10'));
+}
+
+// One-hot encoded `LEDLight` that extends from Const() value.
+// Represent yellow, blue, red and green LED as constant value `00`, `01`,
+// `10`, and `11` respectively.
+class LEDLight extends Const {
+  LEDLight._(int super.value) : super(width: 2);
+  LEDLight.yellow() : this._(bin('00'));
+  LEDLight.blue() : this._(bin('01'));
+  LEDLight.red() : this._(bin('10'));
+  LEDLight.green() : this._(bin('11'));
+}
+
+// Define a class OvenModule that extends ROHD's abstract Module class.
+class OvenModule extends Module {
+  // A public variable with type StateMachine<OvenState> `oven`.
+  // We want to return this variable to the main module for flexibility.
+  // Use `late` to indicate that the value will not be null
+  // and will be assign in the later section.
+  late StateMachine<OvenState> oven;
```

Comment:
This could be private since you have a public getter already.  It would be nice to put the getter next to this variable definition